Core: Fix MXCSR and FPUCW registers on created threads

StevenMiller123 opened the pull request:
Orbis seems to set the flush-to-zero and denormals-are-zero flags of MXCSR, unlike all other platforms, and defaults to a higher floating point precision than Windows.
This PR sets these registers using constants dumped from real hardware.

This fixes the currently Linux-specific crash in base versions of Uncharted 4: A Thief’s End™, as described in https://github.com/shadps4-compatibility/shadps4-game-compatibility/issues/146

Let me know if there are any formatting changes I need to make here, I don't typically use inline assembly.

StevenMiller123 committed:
Set MXCSR on all platforms

All three platforms use the same default MXCSR register value. Use an assembly instruction to set this for all platforms.

StevenMiller123 committed:
Use _mm_setcsr over assembly

Exists on all platforms, so might as well use it.

StevenMiller123 committed:
Clang

StevenMiller123 committed:
Missing include

squidbus commented on src/core/thread.cpp:
```diff
@@ -130,7 +125,11 @@ void NativeThread::Exit() {
+    _mm_setcsr(ORBIS_MXCSR);
 #if _WIN64
+    // Windows needs the FPUCW register set manually.
+    asm volatile ("fldcw %0" : : "m"(ORBIS_FPUCW));
```
I think we might as well set this unconditionally, just in case. It shouldn't add any noticable overhead or anything.

StevenMiller123 committed:
Unconditionally set FPUCW

As per review suggestions.

Commit:
Fix FPUCW and MXCSR registers on Windows

For some reason only Microsoft knows, settings these values in the context doesn't work.
Also their controlfp functions don't seem to work either, so I used assembly instead.
